### PR TITLE
ENH: add TabularMSA.gap_frequencies

### LINKED ISSUE #1089: `TabularMSA`: gap_frequencies
From [RFC](https://github.com/biocore/scikit-bio-rfcs/blob/master/active/001-tabular-msa.md):

- `gap_frequencies(axis='position', relative=False) -> np.ndarray`

   Return vector of gap frequencies along an axis.

### PR DESCRIPTION
**Pull request by jairideout**
Fixes #1089. Tried out Arrange Act Assert (#1149) for these unit tests. I like it!

This is the first method with an `axis` parameter. The RFC specifies that `'sequence'` and `0` are equivalent, likewise with `'position'` and `1`. However, our notion of `axis` differs from numpy:

```python
In [1]: import numpy as np

In [2]: a = np.array([[1,2,3], [4,5,6]])

In [3]: a
Out[3]:
array([[1, 2, 3],
       [4, 5, 6]])

In [4]: a.sum(axis=0)
Out[4]: array([5, 7, 9])

In [5]: a.sum(axis=1)
Out[5]: array([ 6, 15])

In [6]: from skbio import DNA, TabularMSA

In [7]: msa = TabularMSA([DNA('ACG'), DNA('A--')])

In [8]: msa.gap_frequencies(axis=0)
Out[8]: array([0, 2])

In [9]: msa.gap_frequencies(axis=1)
Out[9]: array([0, 1, 1])
```

Is scikit-bio's notion of `axis` correct or should we be matching numpy? Intuitively our API makes more sense to me but not matching numpy may be a mistake...

### CODE REVIEW

**Comment by ebolyen** [comment]
It looks like our semantics are backwards. `axis` means *across* not *along*. 
Here is an example in `pandas`
```
In [9]: df
Out[9]: 
   1  2
0  1  5
1  2  6
2  3  7

In [10]: df.sum(axis='columns')
Out[10]: 
0     6
1     8
2    10
dtype: int64

In [11]: df.sum(axis='rows')
Out[11]: 
1     6
2    18
dtype: int64

```

So positional gap frequencies would be `gap_frequencies(axis='sequence')` because we are finding the gap frequencies across the sequence axis.

Since we intend to interoperate closely with `pandas` and `numpy` (and maybe `xray`!) we need to make sure we match the semantics of `axis` which means *across*.

It also makes sense why `numpy` had to use this meaning in the first place. Consider a 5 dimensional array, if we wanted to sum across a single axis we can either specify that single axis, or state that we want to sum *along* all of the other dimensions. It just looks a little backwards in `2d`.

**Comment by jairideout** [comment]
Thanks @ebolyen! That makes sense, will update shortly.

**Comment by jairideout** [comment]
How about removing the integer axis names (`0` and `1`) in the spirit of named axes? From DS4DS it sounded like numpy might support named axes in the future, and devs frowned on code that used numbers instead of descriptive names. We could always add integer support in the future if users ask for it without breaking backward-compatibility, and in the meantime, scikit-bio *user* code will be more readable.

**Comment by jairideout** [comment]
@wasade what do you think re: having only named axis arguments? (See my previous comment for context)

**Comment by wasade** [comment]
I really like descriptive axis labels

**Comment by jairideout** [comment]
Cool! The specific question is: do we want to support both named axes and integer axes, or just named axes?

**Comment by wasade** [comment]
Oh, I see. I guess the benefit of integer axes is it maps onto numpy right now. I don't really see any other benefit except performance within a conditional as an integer test is faster than a string comparison. But, numpy is (likely) shifting away from unnamed axes, and I don't know if the performance difference warrants the semantic confusion. I think supporting multiple ways of describing the axes can lead to annoying bugs, although I admit there were times I wished it were possible to use the numbered axes in BIOM to reduce line length.

**Comment by jairideout** [comment]
Awesome, thanks for the input! Sounds like @ebolyen, you, and I are on the fence on this one. I'm leaning towards keeping integer and string axis support so that there's a convenient, short way to reference axes via integers. In all documentation examples we'll use the named (string) version of the axes to encourage users to do the same.

Anyone else have an opinion?

fence

**Comment by jairideout** [comment]
@ebolyen ready for review

**Comment by gregcaporaso** [comment]
> I'm leaning towards keeping integer and string axis support

I think this is a good way to go too (and we can always revisit before we leave ``experimental``).

**Comment by jairideout** [comment]
:+1:

**Comment by ebolyen** [comment]
@jairideout merge conflicts

**Comment by jairideout** [comment]
merge conflicts resolved

File: skbio/alignment/_tabular_msa.py [hunk 1]
```diff
@@ -484,6 +484,97 @@ def __ne__(self, other):
...
+        array([ 0.  ,  0.25,  0.75])
+
+        Compute frequency of gap characters for each sequence (i.e., *across*
+        the position axis):
+
+        >>> msa.gap_frequencies(axis='position')
+        array([0, 2, 1, 1])
+
+        """
+        if axis == 'sequence' or axis == 0:
```
Comment (ebolyen) [inline skbio/alignment/_tabular_msa.py]: Is there any value to encoding this logic (and the validation) into a helper method like `_is_sequence_axis(axis)`?

The the code could be:
```python
if self._is_sequence_axis(axis):
    #stuff
else:
    #other
#The error was already raised by `_is_sequence_axis`
```

Comment (jairideout) [inline skbio/alignment/_tabular_msa.py]: done

**Pull request merged**